components/support/nimbus-fml/src/util/loaders.rs

```diff
@@ -51,11 +53,101 @@ impl Default for LoaderConfig {
     }
 }
 
+/// A FilePath for a file hosted in a GitHub repository with a specified ref.
+#[derive(Clone, Debug)]
+pub struct GitHubRepoFilePath {
+    /// The repository id, i.e,. `owner/repo`.
+    repo_id: String,
+
+    /// The Git ref.
+    git_ref: String,
+
+    /// A Url, which is only used so that we can re-use Url::join for paths
+    /// inside the repository.
+    ///
+    /// The URL scheme and host will be "bogus" and should not be referenced.
+    ///
+    /// Instead of this, you probably want [`Self::path()`] or
+    /// [`Self::default_download_url()`].
+    url: Url,
+}
+
+impl GitHubRepoFilePath {
+    pub fn new(repo_id: &str, git_ref: &str) -> Self {
+        Self {
+            repo_id: repo_id.into(),
+            git_ref: git_ref.into(),
+            url: Url::parse("bogus://bogus/").expect("This is a constant, valid URL"),
```

Comment:
Nit: this is obviously not supposed to be valid, but could we have nicer word than "bogus". "unused" perhaps, or "no-need-to-be-http"?